xapian-bindings/android/CROSS_COMPILE

```diff
@@ -55,14 +55,50 @@ make install
 
 The above configure will build xapian statically with position independent code.
 
+Generating the bindings for android
+===================================
+Now generate the bindings for android as follows:
+
+cd <path_to_xapian>/xapian-bindings/android
+mkdir -p org/xapian
+swig -Werror -I. -I../../xapian-core/include -c++ -java -module Xapian -package org.xapian -outdir org/xapian -o xapian_wrap.cc android.i
+
+This should generate xapian_wrap.cc,xapian_wrap.h and .java files in org/xapian. Now add the following line in org/xapian/XapianJNI.java at start of static block at line no 1114:
+
+System.loadLibrary("xapian_jni");
+
+Save the file. Now generate the libxapian_jni.so by doing the following:
+
+export CXX=$TOOLCHAIN/bin/<host>-g++
+$CXX -c xapian_wrap.cc -I../../xapian-core/include
+$CXX -shared -o built/lib/libxapian_jni.so -Wl,-whole-archive libxapian-1.3.a  -Wl,-no-whole-archive xapian_wrap.o -lz -luuid -lc
+
+This will generate libxapian_jni.so in built/lib/libxapian_jni.so. Now compile all the .java class using :
+
+javac -source 1.7 -target 1.7 -bootclasspath <path_to_jvm>/<your_jdk>/jre/lib/rt.jar org/xapian/*.java
+
+<path_to_jvm> is the jvm installation path eg.usr/lib on Ubuntu
+The 'source' and 'target' flags are required for android compactibility if you have java 1.8 or above. 
+
+
+Now create a proper file structure for .aar for android by:
+
+mkdir -p aar/jni/<proper_architecture> aar/res
+jar -cf aar/classes.jar org/xapian/*.class
+cp built/lib/libxapian_jni.so aar/jni/<proper_architecture>/
+cd aar
+jar -cf ../built/xapianlib.aar *
```

Comment:
The next stage of this should probably be to turn it all into make rules within the bindings automake system, so people can just type `make` rather than having to cut and paste commands. I'd guess that we should not try to build Android by default, and maybe should require `TOOLCHAIN` (probably as `ANDROID_TOOLCHAIN`) to be passed into the configure system.